katas/content/qec_shor/index.md

```diff
@@ -0,0 +1,371 @@
+From the computation point of view, the presence of noise in the quantum system means that its state can suffer from random errors, and thus end up different from the state we're relying on to do our computations. 
+This makes computations unreliable very fast, since the effects of noise accumulate quickly to make computation results effectively random.
+
+Before we dive into dealing with noise in quantum systems, let's take a quick look at how we do that for classical systems.
+
+The model used for analyzing classical noise is called *binary symmetric channel*, in which classical bits sent through the channel are transmitted correctly with probability $1-p$ and flipped with probability $p$.
+In this scenario, the information sent through the channel can be protected against the effects of the noise using *repetition code*:
+
+- On the sender side, we replace each bit we want to send with three copies of itself:
+$$0 \rightarrow 000, 1 \rightarrow 111$$
+- On the receiver side, we decode the original bit by majority vote: 
+$$000, 100, 010, 001 \rightarrow 0$$
+$$111, 011, 101, 110 \rightarrow 1$$
+
+What is the probability of this scheme failure, that is, the value of the message bit changing after it was sent through the channel? 
+Majority vote allows for one error on any bit to happen without affecting the decoding outcome, so it would take two or three errors happening on individual bits for decoding to produce an incorrect result. The probability of this happening is $3p^2(1-p) + p^3 = 3p^2 - 2p^3$. If we compare this with the probability of an individual bit transmission failing $p$, we can see that using repetition code yields higher success probability, as long as $p < \frac12$. We can improve success probability further by increasing the number of repetitions we use to encode each bit: $5$ repetitions allow us to detect and correct $2$ errors, $7$ repetitions - $3$ errors, and so on.
+
+> This noise model is useful not only for describing noisy communication channels, but also for memory - any classical system that introduces errors in information when it is left on its own, as opposed to systems that introduce errors during information manipulation. Indeed, we assume that no errors are introduced as we copy the bits during encoding or read and compare their values during decoding.
+
+The main idea of quantum error correction is the same as that for classical error correction: encode information with enough redundancy that we can recover the message even from the noisy transmission results.
+Dealing with the noise in quantum systems is more challenging than in classical systems, though, due to the limitations imposed by their nature:
+
+- **No cloning**: We cannot replicate repetition code for quantum systems in a straightforward manner, by duplicating the quantum state several times, since the no-cloning theorem prohibits that.
```

Comment:
```suggestion
- **No cloning**: We cannot replicate the repetition code for quantum systems in a straightforward manner, by duplicating the quantum state several times, since the no-cloning theorem prohibits that.
```